File: Python/Niveau 1/6 – Structures avancées/Protection du village.py
Maisons = int(input())
absMax = 1
absMin = 1000000
ordMax = 1
ordMin = 1000000

for i in range(Maisons):
   abss = int(input())
   ords = int(input())
   
   if abss > absMax:
      absMax = abss
   elif abss < absMin:
      absMin = abss
   if ords > ordMax:
      ordMax = ords
   elif ords < ordMin:
      ordMin = ords
  
larg = absMax - absMin
longue = ordMax - ordMin

if absMin > absMax:
   larg = absMin - absMax
   print((2 * larg - 1) + (2 * longue - 1))
else:
   print((2 * larg) + (2 * longue))

File: Pascal/Niveau 1/6 – Structures avancées/Protection du village.pas
program Solution;
var
    Houses, absMin, absMax, ordMin, ordMax, abss, ords, i, larg, long: LongInt;
begin
    read(Houses);
    absMax := 1;
    ordMax := 1;
    absMin := 1000000;
    ordMin := 1000000;
    for i := 1 to Houses do
    begin
        read(abss, ords);
        if abss > absMax then
        begin
            absMax := abss;
        end
        else if abss < absMin then
        begin
            absMin := abss;
        end;
        if ords > ordMax then
        begin
           ordMax := ords; 
        end
        else if ords < ordMin then
        begin
            ordMin := ords;
        end;
    end;
    larg := absMax - absMin;
    long := ordMax - ordMin;
    if absMin > absMax then
    begin
        larg := absMin - absMax;
        writeln((2 * larg - 1) + (2 * long - 1));
    end
    else
    begin
        writeln((2 * larg) + (2 * long));
    end;
end.